### PR TITLE
Tweaks to ReferencePolicy GEP

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind gep

**What this PR does / why we need it**:
This is a quick follow up to #711 that addresses a few leftover comments on that PR + does some minor cleanup.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

Deploy Preview: https://deploy-preview-722--kubernetes-sigs-gateway-api.netlify.app/geps/gep-709/

### CODE REVIEW

**Comment by robscott** [comment]
I think we're pretty close to consensus on this, but will hold as a precaution.

/hold

File: site-src/geps/gep-709.md
Comment (robscott) [inline site-src/geps/gep-709.md:91]: In the previous PR, a suggestion was made to rename this to [InboundReferencePolicy](https://github.com/kubernetes-sigs/gateway-api/pull/711#discussion_r670025962). Wanted to pick up the conversation in this PR. I think I slightly prefer the shorter `ReferencePolicy` name, but admit that `InboundReferencePolicy` would make room for something like `OutboundReferencePolicy`. That might allow admins to restrict where Gateways in their namespace could forward to. Thoughts?

/cc @youngnick @bowei @danehans @hbagdi @jpeach

File: mkdocs.yml
Comment (robscott) [inline mkdocs.yml:37]: Adding full GEP names here could end up being a bit much for our existing nav structure, would anyone prefer a single GEP index page that then linked to each individual GEP? I don't feel strongly either way, but wanted to start with something. (https://deploy-preview-722--kubernetes-sigs-gateway-api.netlify.app/geps/gep-709/)

File: site-src/geps/gep-709.md
Comment (jpeach) [reply site-src/geps/gep-709.md:91]: Yeh, my preference is also for a single ReferencePolicy with From+To.

Comment (robscott) [reply site-src/geps/gep-709.md:91]: Yeah as I've been thinking about it more, it would not be that difficult to expand ReferencePolicy to cover both inbound and outbound references. I don't think that's something we actually want to do, but the only required change would be adding `namespace` to the `to` struct. The larger hurdle would be the complexity for users in trying to restrict/allow both inbound and outbound references.

Comment (robscott) [reply site-src/geps/gep-709.md:91]: As discussed in community meeting today, we believe it is unlikely that we will ever have a need for an OutboundReferencePolicy. Given that, the preference is to keep the simpler `ReferencePolicy` name. I'll update the comments around ReferencePolicy to make it clearer that it refers to inbound references.

Comment (robscott) [reply site-src/geps/gep-709.md]: I've updated the wording around this, PTAL

**Comment by robscott** [comment]
Thanks to everyone for the great feedback around this! Based on the discussion at the community meeting today, it sounds like this was pretty close to consensus with no major objections. Given that, I'm going to remove the hold and move this back to a place where an LGTM can merge this in.

/hold cancel

**Comment by youngnick** [comment]
I did the last one, so I should probably leave this to someone else? :)

**Review by howardjohn — APPROVED** [review]

**Comment by robscott** [comment]
ping @hbagdi @jpeach @bowei for final review

File: site-src/geps/gep-709.md
Comment (jpeach) [inline site-src/geps/gep-709.md:234]: Should the type also change to `string*`?

Comment (robscott) [reply site-src/geps/gep-709.md:234]: Updated, thanks!

**Review by jpeach — APPROVED** [review]
Looks good. Omitting `/lgtm` until other reviewers can take a look.

**Pull request merged**
